src/SonarScanner.MSBuild.PreProcessor/WebClientDownloader.cs

```diff
@@ -34,7 +35,7 @@ public class WebClientDownloader : IDownloader
         private readonly ILogger logger;
         private readonly HttpClient client;
 
-        public WebClientDownloader(string userName, string password, ILogger logger)
+        public WebClientDownloader(string userName, string password, string clientCertPath, string clientCertPassword, ILogger logger)
```

Comment:
As we have many nulls here, one better thing would be to pass those 2 parameters after the logger, and put a default null value, it will ease the reading and avoid those nulls on the tests.